bench/workgen/workgen.cpp

```diff
@@ -1902,10 +1909,25 @@ ThreadRunner::op_run(Operation *op)
 
         do {
             // Wait for transactions to complete before stopping.
+            bool has_mirror;
             for (int count = 0; (!_stop || _in_transaction) && count < op->_repeatgroup; count++) {
                 for (std::vector<Operation>::iterator i = op->_group->begin();
                      i != op->_group->end(); i++) {
-                    WT_ERR(op_run_setup(&*i));
+                    auto [table_uri, tint] = op_get_table(&*i);
+                    {
+                        const std::shared_lock lock(*_icontext->_dyn_mutex);
+                        has_mirror = _icontext->_dyn_table_runtime.count(tint) > 0 &&
+                          _icontext->_dyn_table_runtime[tint].has_mirror();
+                    }
+                    /*
+                     * If a mirrored operation has started and there is no active transaction, it
+                     * means it has been interrupted, quit.
+                     */
+                    if ((*i)._random_table && has_mirror && (i != op->_group->begin()) &&
```

Comment:
Can we check that this can't also happen when the operation has just started? 
If the `i != op->_group->begin()` causes us to miss a potential gap.

Reply:
As discussed offline, the first operation (when `i == op->_group->begin()`) has not necessarily started a transaction, therefore it is better to have this test once the first operation has been processed.

Reply:
It sounds like this if statement should be an assert rather than a goto err @etienneptl what do you think

Reply:
We want to abort the list of operations if something went wrong in the middle. One way to exit the loop is to jump to the `err` label.